core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -238,11 +234,7 @@ private[spark] class ExternalSorter[K, V, C](
       }
     } else {
       if (maybeSpill(buffer, buffer.estimateSize())) {
-        buffer = if (useSerializedPairBuffer) {
```

Comment:
This `if` statement for figuring out which buffer to use was replicated in two places, so I refactored it into a `newBuffer()` method, which is defined at the top of this file.  This allowed me to reduce the scope / visibility of some variables like `metaInitialRecords` and `kvChunkSize`.